joomla-framework/database
# New isConnectionEncryptionSupported method

### Summary of Changes

Continuing https://github.com/joomla-framework/database/pull/177.
To check if the there is server side support to TLS connection to the database this PR adds a new method `isConnectionEncryptionSupported` in MySQLi/PDO MySQL/PDO PostgreSQL drivers

### Testing Instructions

1. Code review
2. Use joomla 4 latest + this changes
3. Prepare your database server to support encrypted conenction
<details>
 <summary>For MySQLi / PDO MySQL</summary>
- Add in /etc/my.cnf (replace for instance, with your public https certificates/private key)

```ini
[mysqld]
; [...] more mysql/mariadb server configuration options
ssl        = 1
ssl_cert   = /etc/pki/tls/certs/fullchain.cer ; certificate + intermediates
ssl_ca     = /etc/pki/tls/certs/ca-bundle.crt
ssl_key    = /etc/pki/tls/private/yourdomain.key
ssl_cipher = kECDHE+aECDSA+AESGCM+AES128:kECDHE+aECDSA+AESGCM+AES256:kECDHE+aECDSA+AES128+SHA:kECDHE+aECDSA+AES256+SHA:kDHE+aECDSA+AES128:kDHE+aECDSA+AES256:kECDHE+aRSA+AESGCM+AES128:kECDHE+aRSA+AESGCM+AES256:kECDHE+aRSA+AES128+SHA:kECDHE+aRSA+AES256+SHA:kDHE+aRSA+AESGCM+AES128:kDHE+aRSA+AESGCM+AES256:kDHE+aRSA+AES128:kDHE+aRSA+AES256:kRSA+aRSA+AESGCM+AES128:kRSA+aRSA+AESGCM+AES256:kRSA+aRSA+AES128+SHA:kRSA+aRSA+AES256+SHA
```
- restart the server `systemctl restart mariadb` (or `systemctl restart  mysqld`)
</details>

<details>
 <summary>For PostgreSQL </summary>
- Add in/var/lib/pgsql/data/postgresql.conf (or /var/lib/pgsql/[VERSION]/data/postgresql.conf) (replace for instance, with your public https certificates/private key)

```ini
ssl = on
ssl_ca_file = '/etc/pki/tls/certs/ca-bundle.crt'
ssl_cert_file = '/etc/pki/tls/certs/fullchain.cer' # certificate + intermediates
ssl_key_file = '/etc/pki/tls/private/yourdomain.key'
ssl_ciphers = 'kECDHE+aECDSA+AESGCM+AES128:kECDHE+aECDSA+AESGCM+AES256:kECDHE+aECDSA+AES128+SHA:kECDHE+aECDSA+AES256+SHA:kDHE+aECDSA+AES128:kDHE+aECDSA+AES256:kECDHE+aRSA+AESGCM+AES128:kECDHE+aRSA+AESGCM+AES256:kECDHE+aRSA+AES128+SHA:kECDHE+aRSA+AES256+SHA:kDHE+aRSA+AESGCM+AES128:kDHE+aRSA+AESGCM+AES256:kDHE+aRSA+AES128:kDHE+aRSA+AES256:kRSA+aRSA+AESGCM+AES128:kRSA+aRSA+AESGCM+AES256:kRSA+aRSA+AES128+SHA:kRSA+aRSA+AES256+SHA'
```
- restart the server `systemctl restart postgresql` (or `systemctl restart  postgresql-[VERSION]`)
</details>
4. Call the new method enabling/disabling ssl on the server, if is not available the method should give `false`, if available should give  `true`. Something like:

`echo $db->isConnectionEncryptionSupported();`


### Documentation Changes Required

None.

@richard67 this time i have setup a postgresql database to test, but still no expert in this.
